dht.go

```diff
@@ -216,19 +218,12 @@ func New(ctx context.Context, h host.Host, options ...Option) (*IpfsDHT, error)
 	// handle providers
 	dht.proc.AddChild(dht.ProviderManager.Process())
 
-	if err := dht.rtRefreshManager.Start(); err != nil {
-		return nil, err
-	}
+	dht.proc.Go(dht.populatePeers)
```

Comment:
`dht.populatePeers` to be in its own goroutine nothing about it needs to happen asynchronously except for `dht.proc.Go(dht.fixLowPeersRoutine)` which already happens asynchronously.

If you just pull that whole function and put it back here things should be fine. We should cleanup the `New` and `makeDHT` functions, but we can just do that in another PR that's purely a refactor.

Note: this also means that making `disableFixLowPeers` a field is no longer necessary

Reply:
I'm confused because it appears that @petar is suggesting the opposite here:

https://github.com/libp2p/go-libp2p-kad-dht/pull/694#discussion_r490360285

Reply:
Misunderstanding. I was saying that it didn't need to be in it's own goroutine if we didn't block on running fixLowPeers before starting the rtRefreshManager. However, even though it's not a perfect solution for now we can do this per https://github.com/libp2p/go-libp2p-kad-dht/pull/694#discussion_r508613135.